recipes-sota/rac/files/client.toml

```diff
@@ -9,6 +9,8 @@ client_key_path = "/var/sota/import/pkey.pem"
 ssh_private_key_path = "/home/torizon/run/rac/device-key-01.sec"
 unprivileged_user_group = "torizon:torizon"
 local_tuf_repo_path = "/home/torizon/run/rac/uptane-repo"
+poll_timeout = { secs = 30, nanos = 0 }
+validation_poll_timeout = { secs = 30, nanos = 0 }
```

Comment:
> does it mean it is always performed more like a secondary method independent of MQTT?

This. It's performed in a fixed interval (set by `poll_timeout`) so it doesn't check if the MQTT connection is up or anything, just whether it hasn't received any MQTT messages via D-Bus for `poll_timeout` seconds.
The `validation_poll_timeout` is the same logic but for the validation of the session after it has already been established.

> If the polling is always performed even when the MQTT connection is up, I'd ask if this polling interval is appropriate as a default. I imagine the setting here will represent a least a couple of megabytes/month/device, right?

Yes, you're correct. It's miles better than it was before (polling every 3s, which was very bandwidth-heavy) but may still be too much for some users. My original intention was to completely disable the HTTP polling by adding a new option `event_mode_only`, but that got scraped - for now - after a meeting with the cloud team. I think having this longer poll time until we land those features and the MQTT feature stabilizes in the cloud is a good strategy.

Let me know what you think!

Reply:
Is is possible to verify if MQTT is working properly to actually avoid this fallback? If so, this could become an improvement ticket, since it seems we still have the problem we were trying to solve initially.

Reply:
We know that data usage is a concern.

A suggestion:

* Use a reasonable default value, and implement it in a way it can easily be overridden in /etc so users can capture changes with TCB
* Create a section on https://developer.toradex.com/torizon/torizon-platform/remote-access/ talking about data usage, how to tweak default values, and what are the trade-offs between longer polling time and service availability

I'm just jumping in because I feel like this usage requirement may impact the implementation, so rather sooner than later to avoid rework, etc.

Reply:
> Is is possible to verify if MQTT is working properly to actually avoid this fallback?

Actually, not easily because of the architecture. The code that would be responsible for this check would be `tzn-mqtt`, not `rac`. So there would need to maybe be a new message from `tzn-mqtt` basically saying "cannot reach MQTT broker, can you please use HTTP k thx bye" to `rac` (and `aktualizr` for that matter).

Reply:
Gotcha. If it possible to do, I think it would be nice to have this improvement captured in a ticket, and meanwhile we have this "fallback always on" approach.

Reply:
@leograba I guess this is more of a decision for you then: should we do this work now or merge as-is - with the corrections maintainers asked for, naturally - (which I'm in favor of because it's a beta feature and needs testing and it's incrementally better than the previous state)?

Reply:
Yes, merge as-is sounds good. No need to hold improvements

Reply:
Ok, I'll capture the improvements in tickets.